src/main/java/org/elasticsearch/index/query/QueryValidationException.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.elasticsearch.action.ActionRequestValidationException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This exception can be used to indicate various reasons why validation of a query has failed.
+ */
+public class QueryValidationException extends IllegalArgumentException {
+
+    private final List<String> validationErrors = new ArrayList<>();
+
+    public QueryValidationException() {
```

Comment:
maybe make the first validation error mandatory, by using a constructor that accepts an error rather than the default one?